### PR TITLE
Hackweek monday: Simplify scheduler logic

### PR DESCRIPTION
**Pull request by coolo**
Querying all jobs and workers and then doing a NxM assignment in memory is just a dozen times quicker than doing M complex - and also much easier to debug in case of errors as you have easy to follow steps (to be refactored for sure though).

I also removed all failure protection code as I found the scheduler to work better without.

I'm monitoring how it's behaving through out the course of the hackweek, but refactoring and adapting test cases is for another week.

### CODE REVIEW

File: lib/OpenQA/Scheduler/Scheduler.pm [hunk 1]
```diff
@@ -361,30 +314,17 @@ sub schedule {
...
         }
         else {
-            log_debug("Failed sending job '"
-                  . $allocated->{id}
-                  . "' to worker '"
-                  . $allocated->{assigned_worker_id} . "' : "
-                  . pp($res));
-            $failure++
-              if $soft_busy;    # We failed dispatching it. We might be under load, but it's not a big issue
+            log_debug("Failed sending job '" . $allocated->{job} . "' to worker '" . $allocated->{worker});
```
Comment (foursixnine) [inline lib/OpenQA/Scheduler/Scheduler.pm]: I'd promote this message to error.

**Comment by coolo** [comment]
You can see the base load going down throughout hackweek:

cpuusage

File: lib/OpenQA/Scheduler/Scheduler.pm (hunk 1 above)
Comment (coolo) [reply lib/OpenQA/Scheduler/Scheduler.pm]: it's a little less than an error and a little more than debug - but with shared workers it's business as usual that they get 2 jobs assigned at the same time and refuse to accept it.

Comment (foursixnine) [reply lib/OpenQA/Scheduler/Scheduler.pm]: info then, since it's an expected failure

**Comment by coolo** [comment]
superseded by #1729

**Pull request closed without merging**